tests/unittests/test_datasource/test_ec2.py

```diff
@@ -308,21 +418,50 @@ def test_network_config_property_set_dhcp4_on_private_ipv4(self):
             platform_data=self.valid_platform_data,
             sys_cfg={'datasource': {'Ec2': {'strict_id': True}}},
             md={'md': DEFAULT_METADATA})
-        find_fallback_path = (
-            'cloudinit.sources.DataSourceEc2.net.find_fallback_nic')
+        find_fallback_path = M_PATH_NET + 'find_fallback_nic'
         with mock.patch(find_fallback_path) as m_find_fallback:
             m_find_fallback.return_value = 'eth9'
             ds.get_data()
 
         mac1 = '06:17:04:d7:26:0A'  # IPv4 only in DEFAULT_METADATA
-        expected = {'version': 1, 'config': [
-            {'mac_address': '06:17:04:d7:26:0A', 'name': 'eth9',
-             'subnets': [{'type': 'dhcp4'}],
-             'type': 'physical'}]}
-        patch_path = (
-            'cloudinit.sources.DataSourceEc2.net.get_interfaces_by_mac')
-        get_interface_mac_path = (
-            'cloudinit.sources.DataSourceEc2.net.get_interface_mac')
+        expected = {'version': 2, 'ethernets': {'eth9': {
+            'match': {'macaddress': mac1.lower()}, 'set-name': 'eth9',
```

Comment:
why do we have mac1.lower()?   Let's just write mac1 as lower.  If you want to test yaml to netplan rendering of mac address value, do that in a separate unittest.

Reply:
dropped all lower/upper now I believe.

Reply:
the one test case that has to exercise that is test_network_config_property_set_dhcp4_on_private_ipv4
which mocks what mac address is returned by get_interfaces_by_mac (which is uppercase and has to match IMDS macs keys (also uppercase).